### ISSUE TITLE
Cannot use `Add Extra Call Argument` and `Unbind Signal Arguments` at the same time in editor

### ISSUE THREAD

**Issue by jinyangcruise** [godotengine/godot#98686]
### Tested versions

- Reproducible in v4.4.dev3.official [f4af8201b]

### System information

Windows 11, Vulkan API 1.3.280 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3060 Ti

### Issue description

When connect a method to a signal in the editor, we can not use `Add Extra Call Argument` and `Unbind Signal Arguments` at the same time. As long as we set a number ( number > 0) to the `Unbind Signal Arguments`, the Color of the `Add Extra Call Argument` area will become gray and any extra arguments will be discard after confirmation.

They should be able to use at the same time as the code does:
```
func _ready() -> void:
	var callable = test.bind(true, true, true).unbind(1)
	custom_signal.connect(callable)
	custom_signal.emit(1, 2, 3)
```

![屏幕截图 2024-10-31 114631](https://github.com/user-attachments/assets/1af82f34-4f06-4669-ba87-0c7bc5280bfb)
![屏幕截图 2024-10-31 114652](https://github.com/user-attachments/assets/ceaa0d55-d351-4a4c-8440-a66755a91e4b)
<img width="329" alt="屏幕截图 2024-10-31 114702" src="https://github.com/user-attachments/assets/bd65ecf8-88d2-4a03-bf48-dc4c26e4eb9d">

### Steps to reproduce

1. create a 2D node scene and attach a script with the code below
2. connect the custom_signal and add some extra arguments and set the unbind signal argument field

```
extends Node2D

signal custom_signal(a, b, c)

func test(a, b, c, d, e):
	printt(a, b, c, d, e)
	
#func _ready() -> void:
	#var callable = test.bind(true, true, true).unbind(1)
	#custom_signal.connect(callable)
	#custom_signal.emit(1, 2, 3)
```

### Minimal reproduction project (MRP)

NA

**Comment by quinnyo** [comment]
I noticed this today (on v4.4.beta1.official [d33da79d3])

I can't see any reason for it being designed like this. It seems like a mistake to me.

If there is some reason for the current behaviour, it would be great if someone could please let me know. I'm happy to discuss it further!

**Comment by Macksaur** [comment]
#100554 fixes it but I'm not sure why it's not being picked up. I think it needs more 👍?

**Comment by jinyangcruise** [comment]
@quinnyo I guess the reason why the old behavior doesn't support bind and unbind at the same time is that the most common usage of using bind and unbind is only using bind or only using unbind. However we are allowed to use bind and unbind at the same time by code like:
```
@tool
extends EditorScript

func f(a = 0, b = 0, c = 0, d = 0, e = 0, f = 0, g = 0, h = 0):
    return [a, b, c, d, e, f, g, h]

func _run() -> void:
    f.bind(1, 2)
    f.bind(1, 2).unbind(1)
    f.bind(1, 2).unbind(1).bind(3, 4)
    f.bind(1, 2).unbind(1).bind(3, 4).unbind(1)
    
    f.bind(1).bind(2).bind(3).unbind(1)
    f.bind(1).bind(2).unbind(1).bind(3)
    f.bind(1).unbind(1).bind(2).bind(3)
    f.unbind(1).bind(1).bind(2).bind(3)
    
    f.unbind(1).unbind(1).unbind(1).bind(1, 2, 3)
    f.unbind(1).unbind(1).bind(1, 2, 3).unbind(1)
    f.unbind(1).bind(1, 2, 3).unbind(1).unbind(1)
    f.bind(1, 2, 3).unbind(1).unbind(1).unbind(1)
```
The order and count of using bind and unbind are very important and not very simple to clarify by GUI. I guess that's also a main reason why they haven't supported using them at the same time.

Hey, thx for your fixes @Macksaur . I give it a 👍. But I don't know why it's not being picked up.
I notice you say this fix behaves like `source_signal.connect(target_function.bind(args).unbind(n))`,
I'm looking forward to using any kind of combination of order and count of `bind` and `unbind`, maybe by a form of list which can re-order its elements by buttons or drag-and-drop.

**Comment by Macksaur** [comment]
> I'm looking forward to using any kind of combination of order and count of bind and unbind, maybe by a form of list which can re-order its elements by buttons or drag-and-drop.

Can you give an example of a specific problem you're facing that would be solved by arbitrary reordering?

I did it that way because a premade signal has parameters that you cannot get rid of otherwise. They must be removed before allowing the appending of the custom parameters that your target function actually cares about. You ideally shouldn't be adding any further parameters beyond this in the GUI, only to unbind them.

Custom reordering imo is best done entirely in code. The GUI just needs to support the 99% case.

**Comment by jinyangcruise** [comment]
Yes, you are right. I misunderstood the function of `unbind` in this GUI, in fact it's for `Unbind Signal Arguments` not `Unbind method Arguments`. So what I said was invalid.

**Comment by Macksaur** [comment]
It is ok, it is good food for thought. One day I'd like to be able to reorder the functions parameters properly and not use clunky bind/unbind. But small steps first!

**Comment by Rindbee** [comment]
Because it is currently LIFO instead of FIFO, using both at the same time is not very meaningful and may increase the probability of error.

You can try to output the argument list of the receive method after creating the connection.

Especially for `bind()`, it is not recommended to call it multiple times in a row.
